charm/src/charm.py

```diff
@@ -53,6 +54,13 @@ def __init__(self, *args):
             database_name="testflinger_db",
         )
 
+        self._prometheus_scraping = MetricsEndpointProvider(
+            self,
+            relation_name="metrics-endpoint",
+            jobs=[{"static_configs": [{"targets": [f"*:5000"]}]}],
```

Comment:
I think this is what causes the lint failure. You have f"*:5000" but there's no placeholder for it.